apps/devhub/tests/test_utils.py

```diff
@@ -0,0 +1,373 @@
+
+            # If the message is ignored, we should see it counted only in the
+            # ignored message summary, otherwise we should expect to see it
+            # counted only in the main summary.
+            if expected.get('ignored'):
+                assert summaries == (self.SIGNING_SUMMARY, summary)
+            else:
+                assert summaries == (summary, self.SIGNING_SUMMARY)
+
+    def test_matching_message(self):
+        """Test the behavior of matching messages."""
+
+        # Low severity messages are ignored unless flagged as not ignorable.
+        for severity in 'low', 'trivial':
+            self.compare({'id': ('a', 'b', 'c'),
+                          'signing_severity': severity,
+                          'context': ('x', 'y', 'z'),
+                          'file': 'foo.js'},
+                         {},
+                         {'ignored': True,
+                          'matched': True})
+
+            self.compare({'id': ('a', 'b', 'c'),
+                          'signing_severity': severity,
+                          'ignore_duplicates': False,
+                          'context': ('x', 'y', 'z'),
+                          'file': 'foo.js'},
+                         {},
+                         {'ignored': False,
+                          'matched': True})
+
+        # Other severities are ignored only when flagged as ignorable.
+        for severity in 'medium', 'high':
+            self.compare({'id': ('a', 'b', 'c'),
+                          'signing_severity': severity,
+                          'context': ('x', 'y', 'z'),
+                          'file': 'foo.js'},
+                         {},
+                         {'ignored': False,
+                          'matched': True})
+
+            self.compare({'id': ('a', 'b', 'c'),
+                          'signing_severity': severity,
+                          'ignore_duplicates': True,
+                          'context': ('x', 'y', 'z'),
+                          'file': 'foo.js'},
+                         {},
+                         {'ignored': True,
+                          'matched': True})
+
+        # Check that unchecked properties don't matter.
+        self.compare({'id': ('a', 'b', 'c'),
+                      'signing_severity': 'low',
+                      'context': ('x', 'y', 'z'),
+                      'file': 'foo.js'},
+                     {'thing': 'stuff'},
+                     {'ignored': True,
+                      'matched': True})
+
+        # Non-signing messages should be matched, but not annotated for
+        # ignorability.
+        self.compare({'id': ('a', 'b', 'c'),
+                      'context': ('x', 'y', 'z'),
+                      'file': 'foo.js'},
+                     {},
+                     {'matched': True})
+
+    def test_non_matching_messages(self):
+        """Test that messages which should match, don't."""
```

Comment:
Typo? Should it read `which shouldN'T match`? Otherwise I don't understand the docstring